### PR TITLE
Implement writing support for both v2 and legacy tiny formats

### PR DESCRIPTION
**Pull request by DenWav**

### CODE REVIEW

**Review by jamierocks — COMMENTED** [review]
Looks fine, just missing some documentation on some classes.

**Review by modmuss50 — APPROVED** [review]
LGTM 👍 

Many Thanks.

File: build.gradle [hunk 1]
```diff
@@ -1,55 +1,55 @@
...
+		name = 'Fabric'
+		url = 'https://maven.fabricmc.net/'
+	}
 }
 
-task sourcesJar(type: Jar) {
-    classifier 'sources'
-    from sourceSets.main.allJava
+dependencies {
+	implementation 'org.cadixdev:lorenz:0.5.6'
```
Comment (Juuxel) [inline build.gradle]: Isn't Lorenz also an `api` dependency since lorenz-tiny exposes it in method signatures?

File: build.gradle [hunk 2]
```diff
@@ -1,55 +1,55 @@
...
+	api 'net.fabricmc:tiny-mappings-parser:0.3.0+build.17'
```
Comment (jamierocks) [inline build.gradle:31]: Is this a breaking version change actually, if so - should probably bump to 3.x.

**Comment by i509VCB** [comment]
@modmuss50 I believe this is a good opportunity to get Lorenz tiny on github actions as well

**Review by Player3324 — APPROVED** [review]
Looks ok assuming none of the written names/types/descs contain characters that needs escaping per the spec and that this uses the proper lv index for parameter entries.

**Comment by jamierocks** [comment]
> that this uses the proper lv index for parameter entries

Lorenz doesn't provide any guarantees on this, its up to consumers to fill that data appropriately.

File: build.gradle (hunk 1 above)
Comment (DenWav) [reply build.gradle]: Yeah you're right, will update.

File: build.gradle (hunk 2 above)
Comment (DenWav) [reply build.gradle:31]: Is this a breaking change? Didn't realize it, will bump to 3.x.

**Comment by DenWav** [comment]
> Looks ok assuming none of the written names/types/descs contain characters that needs escaping per the spec

Wasn't aware tiny had escaping rules - I suspect for almost all cases that isn't actually an issue. Could probably easily enough add escaping on top of this though.

File: build.gradle (hunk 2 above)
Comment (jamierocks) [reply build.gradle:31]: I'm not sure if it is - perhaps @modmuss50 could chime in?

**Pull request merged**